frontend/src/components/terminal.js

```diff
@@ -1,5 +1,59 @@
-import React from "react";
+import React, { Component } from "react";
 
-export default function Terminal() {
-  return <div />;
+let distanceFromNewLine = 6;
+
+function caretToggle() {
+  const caret = document.getElementsByClassName("term-caret")[0];
+  if (caret.classList.contains("blink")) {
+    caret.classList.remove("blink");
+  } else {
+    caret.classList.add("blink");
+  }
+}
+
+function logKey(e) {
+  const terminalText = document.querySelector("#terminal-text");
+  const character = `${String.fromCharCode(e.keyCode)}`.toLowerCase();
+  const result = terminalText.childNodes[0];
+  if (e.keyCode === 13) {
+    result.nodeValue += "\nroot$ ";
+    distanceFromNewLine = 6;
+  } else if (e.keyCode === 8) {
+    // dont delete if no characters inputted
+    if (distanceFromNewLine > 6) {
+      result.nodeValue = result.nodeValue.substring(
+        0,
+        result.nodeValue.length - 1
+      );
+      distanceFromNewLine -= 1;
+    }
+  } else {
+    result.nodeValue += character;
+    distanceFromNewLine += 1;
+  }
+}
+
+class Terminal extends Component {
```

Comment:
Can we try to only use functional components for this application to have consistency throughout the codebase?